OpenRA.Mods.Common/Traits/Replaceable.cs

```diff
@@ -0,0 +1,47 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Primitives;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class ReplaceableInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[Desc("Identifyer for this replaceable Actor (matched against Replaces in Replacement).")]
+		public readonly string Type = null;
+
+		public object Create(ActorInitializer init) { return new Replaceable(init, this); }
+	}
+
+	public class Replaceable
```

Comment:
It would probably be a good idea to make this a `ConditionalTrait` so that the replacement can be turned on or off by other things. (e.g. modders may want to disable replacement if an actor is EMPed or shielded).

Reply:
What will i need for this? Methods like `EnablePlug` and `DisablePlug`. Anything else?

Reply:
Making `ReplaceableInfo` inherit from `ConditionalTraitInfo` and this inherit `ConditionalTrait<ReplaceableInfo>` plus my other request that replaces `AcceptsReplacement` with a query that includes `IsTraitDisabled` will be enough.